File: khalti-android/src/main/java/com/khalti/checkOut/EBanking/deepLinkReceiver/DeepLinkContract.java
package com.khalti.checkOut.EBanking.deepLinkReceiver;

import java.util.HashMap;

import com.khalti.checkOut.api.Config;

interface DeepLinkContract {
    interface View {
        HashMap<String, Object> receiveEBankingData();

        Config getConfigFromFile();

        void closeDeepLink();

        void setListener(Listener listener);
    }

    interface Listener {
        void receiveEBankingData();
    }
}

File: khalti-android/src/main/java/com/khalti/checkOut/ebanking/deepLinkReceiver/DeepLinkContract.kt
package com.khalti.checkOut.ebanking.deepLinkReceiver

import com.khalti.base.LifeCycle
import java.util.HashMap

import com.khalti.checkOut.helper.Config
import com.khalti.checkOut.service.ConfigServiceComm

interface DeepLinkContract {
    interface View {

        val configFromFile: Config

        val configFromService: Config?

        fun receiveEBankingData(): Map<String, Any>?

        fun closeDeepLink()

        fun setPresenter(presenter: Presenter)
    }

    interface Presenter : LifeCycle
}
